### PR TITLE
feat(list-header): add ionic theme styles to ionic-link and labels

### PR DESCRIPTION
**Pull request by thetaPC**
Issue number: internal

---------

## What is the current behavior?

List header does not have any `ionic` theme styles for `ion-label` and `a.ionic-link`.

## What is the new behavior?

- Added styles to `ion-label` and `a.ionic-link` when they're inside the `ion-list-header` and the theme is `ionic`
- Added tests

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

Links within the list header should have the `.ionic-link` in order to get the correct states and color.

[Preview](https://ionic-framework-git-rou-10761-ionic1.vercel.app/src/components/list-header/test/basic?ionic:theme=ionic)

### CODE REVIEW

**Review by JoaoFerreira-FrontEnd — APPROVED** [review]

File: core/src/components/item/item.ionic.vars.scss [hunk 1]
```diff
@@ -0,0 +1,5 @@
+/// @prop - Padding start for the item content
+$item-ionic-padding-start: 16px;
+
+/// @prop - Padding end for the item content
+$item-ionic-padding-end: 16px;
```
Comment (brandyscarney) [inline core/src/components/item/item.ionic.vars.scss:5]:
```suggestion
/// @prop - Padding start for the item content
$item-ionic-padding-start: globals.$ionic-space-400;

/// @prop - Padding end for the item content
$item-ionic-padding-end: globals.$ionic-space-400;
```

File: core/src/components/list-header/list-header.ionic.scss [hunk 2]
```diff
@@ -0,0 +1,55 @@
+@use "../../themes/ionic/ionic.globals.scss" as globals;
+@import "./list-header";
+@import "../item/item.ionic.vars";
+
+:host {
+  --background: transparent;
+
+  min-height: 58px;
```
Comment (brandyscarney) [inline core/src/components/list-header/list-header.ionic.scss:8]: Where is this coming from?

**Review by brandyscarney — CHANGES_REQUESTED** [review]

File: core/src/components/list-header/list-header.ionic.scss (hunk 2 above)
Comment (thetaPC) [reply core/src/components/list-header/list-header.ionic.scss:8]: It comes from Figma when inspecting the list header.
Screenshot 2024-06-25 at 9 03 15 AM

File: core/src/components/list-header/list-header.ionic.scss [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  font-weight: globals.$ionic-font-weight-regular;
+}
+
+// Link
+::slotted(a.ionic-link) {
+  // Needs to be !important to override the default link color
+  // that is set by the ionic theme at a global level.
+  //
+  /* stylelint-disable declaration-no-important */
+  color: #{globals.$ionic-color-primary-base} !important;
```
Comment (brandyscarney) [inline core/src/components/list-header/list-header.ionic.scss:54]:
```suggestion
  color: globals.$ionic-color-primary-base !important;
```

**Review by brandyscarney — APPROVED** [review]

**Pull request merged**